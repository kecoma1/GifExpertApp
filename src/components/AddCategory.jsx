import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState('Look for a gif brah');

    // Handling the input
    const handleInputChange = e => setInputValue(e.target.value);

    // Preventing the default behavior of the submit
    const handleSubmit = e => {
        e.preventDefault();
    
        if (inputValue.trim().length > 2) {
            setCategories( c => [inputValue, ...c] );
            setInputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <h2>AddCategory</h2>
            <input 
                type="text"
                value={inputValue}
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
