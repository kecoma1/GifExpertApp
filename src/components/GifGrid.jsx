import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data, loading } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>

            { loading && <p>Loading</p> }

            <div className='card-grid'>
                {
                    data.map( img => 
                        <GifGridItem 
                        key={img.id}
                        { ...img }/>
                    )
                }
            </div>
        </>
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}
