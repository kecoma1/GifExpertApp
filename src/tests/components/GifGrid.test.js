import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests on <GifGrid />:', () => {


    test('Should show the component correctly', () => {
        const gifs = [{
            id: 'aaa',
            url: 'https://ladlffa//ananv',
            title: 'bbb'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category='Hola' /> );

        expect( wrapper ).toMatchSnapshot();
    });


    test('Should display items when images are loaded with useFetchGifs', () => {
        //const wrapper = shallow( <GifGrid category='Hola' /> );
        const gifs = [{
            id: 'aaa',
            url: 'https://ladlffa//ananv',
            title: 'bbb'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        const wrapper = shallow( <GifGrid category='Hola' /> );

        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });

});