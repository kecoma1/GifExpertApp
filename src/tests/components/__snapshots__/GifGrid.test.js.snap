// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGrid />: Should show the component correctly 1`] = `
<Fragment>
  <h3>
    Hola
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="aaa"
      key="aaa"
      title="bbb"
      url="https://ladlffa//ananv"
    />
  </div>
</Fragment>
`;
